assets/wizards/setup/views/settings/index.js

```diff
@@ -8,48 +8,134 @@ import apiFetch from '@wordpress/api-fetch';
 			} )
 			.catch( setError );
 	}, [] );
 
+	const updateProfile = () =>
+		apiFetch( {
+			path: '/newspack/v1/profile/',
+			method: 'POST',
+			data: { profile: profileData },
+		} );
+
+	useEffect( () => {
+		if ( typeof profileData.site_title === 'string' ) {
+			document.title = pageTitleTemplate.replace( '__SITE_TITLE__', profileData.site_title );
+		}
+	}, [ profileData.site_title ] );
+
+	const renderSetting = ( { options, label, key, type, placeholder, className } ) => {
+		if ( options ) {
+			return (
+				<SelectControl
+					disabled={ isLoading }
+					label={ label }
+					value={ profileData[ key ] }
+					onChange={ updateProfileData( key ) }
+					options={ options }
+					className={ className }
+				/>
+			);
+		}
+		if ( type === 'image' ) {
+			return (
+				<ImageUpload
+					label={ label }
+					style={ { width: '136px', height: '136px' } }
+					image={ profileData[ key ] }
+					info={ __(
+						'The Site Icon is used as a browser and app icon for your site. Icons must be square, and at least 512 pixels wide and tall.',
+						'newspack'
+					) }
+					onChange={ updateProfileData( key ) }
+				/>
+			);
+		}
+		return (
+			<TextControl
+				disabled={ isLoading }
+				label={ label }
+				value={ profileData[ key ] || '' }
+				onChange={ updateProfileData( key ) }
+				placeholder={ placeholder }
+				className={ className }
+			/>
+		);
+	};
+
 	return (
 		<Fragment>
-			<SelectControl
-				disabled={ isLoading }
-				label={ __( 'Where is your business based?' ) }
-				value={ profileData.country }
-				onChange={ updateProfileData( 'country' ) }
-				options={ countries }
+			<SectionHeader
+				title={ __( 'Site profile', 'newspack' ) }
+				description={ __( 'Add and manage the basic information', 'newspack' ) }
 			/>
-			<SelectControl
-				disabled={ isLoading }
-				label={ __( 'Currency' ) }
-				value={ profileData.currency }
-				onChange={ updateProfileData( 'currency' ) }
-				options={ currencies }
+			<Grid columns={ 3 } gutter={ 32 } className="newspack-site-profile">
+				{ renderSetting( {
+					key: 'site_icon',
+					label: __( 'Site Icon', 'newspack' ),
+					type: 'image',
+				} ) }
+				<Card noBorder>
+					{ renderSetting( { key: 'site_title', label: __( 'Site Title', 'newspack' ) } ) }
+					{ renderSetting( { key: 'tagline', label: __( 'Tagline', 'newspack' ) } ) }
+				</Card>
+				<Card noBorder>
+					{ renderSetting( {
+						options: countries,
+						key: 'countrystate',
+						label: __( 'Country', 'newspack' ),
+					} ) }
+					{ renderSetting( { options: currencies, key: 'currency', label: __( 'Currency' ) } ) }
+				</Card>
+			</Grid>
+			<SectionHeader
+				title={ __( 'Social accounts', 'newspack' ) }
+				description={ __( 'Allow visitors to quickly access your social profiles', 'newspack' ) }
 			/>
+			<Grid columns={ 3 } gutter={ 32 } rowGap={ 16 }>
+				{ wpseoFields.map( seoField => (
+					<Fragment key={ seoField.key }>
```

Comment:
Is `Fragment` necessary since it only has one child?

Reply:
_Something_ has to have the `key` prop. The key could be passed to the component rendered by `renderSetting`, but I think it's cleaner this way.

Reply:
Understood—I was thinking if `renderSetting` were a component, then that would get the `key`.